### ISSUE TITLE
Sprint March 4 to March 15, 2024

### ISSUE THREAD

**Issue by mariusandra** [PostHog/posthog#20598]
# Global Sprint Planning

## 3 things that might take us down

1. 🍍 
2. 
3. 

# Team sprint planning

For your team sprint planning copy this template into a comment below for each team.

```
# Team ___

**Support hero:** ___

## Retro

<!-- Grab the high and low priority items from last time and add whether that item was completed or not -->

- 

## Hang over items from previous sprint

<!-- For each item, decide to re-prioritise (and add below) or deprioritise -->

- Item 1. prioritised/deprioritise

## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red


### High priority

-

### Low priority / side quests

-

```

**Comment by mariusandra** [comment]
# Team Product Analytics

**Support hero:** 
- Tom

## Retro

We blew some scheduled plans out the water and focused on support related activities.

- Michael worked on Dashboard copying, which morphed into environments
- Julian worked on making CSV exports reliable again, and on insight/dashboard frontend bugs.
- Tom did all manner of support
- Still we had the HogQL rewrite happening in parallel (Thomas and Marius)
  - 🟡 Trends - TBD
  - 🟡 Funnels - TBD
  - 🟡 Paths - TBD

## OKR Q1 2024

1. HogQL & Data Exploration (Julian, Marius)
   - 🟡 Convert ALL our insights to use HogQL as their base.
   - 🟡 Write great docs for HogQL and data exploration nodes.
   - 🔴 Improve the type system and get rid of `assumeNotNull`.
   - 🔴 Proactively monitor query performance.

2. Product Analytics frontend (Thomas, Michael)
   - 🟡 Clean things up now that we have PostHog 3000 and HogQL everywhere.
   - 🟡 Work on the universal "Explore" view.

3. BI (Tom)
   - 🟡 Support non-event data sources on HogQL insights.
   - 🟡 Build the new querying experience, and/or integrate it into the "Explore" view.

## Next sprint

**TODO. This is copied from last time. Will edit during standup.**

- @Gilbert09 
  - Support
  - Linking external data warehouse tables with joins
- @thmsobrmlr 
  - [HogQL funnels](https://github.com/PostHog/meta/issues/130)
  - HogQL paths last step
- @mariusandra 
  - HogQL trends and persons (if still todo)
  - Follow up on type system changes
  - Better JSON handling in HogQL
  - Filters refactor (move the frontend to queries-only)
- @Twixes 
  - Copying dashboards between projects
  - Performance optimisations into HogQL
- @webjunkie 
  - Async queries (celery autoscaling?)
  - Django 4.2 upgrade
  - API docs (get them into posthog.com)
  - Support rollover
  - Debounced filters in the UI

**Comment by pauldambra** [comment]
# Team Replay-i

**Support hero:** Manoel

## Retro

- lost so much time to this rubbish cough 👎  @pauldambra 

## Hang over items from previous sprint

### main story

- iOS session replay have first captured recording @marandaneto 
- 🔴  5 Open Source Android apps tested with android replay @pauldambra 
    - basically a 1 week sprint for me 😷
    - clearing my plate so I can focus on this
    - this will carry forward 
- get "show similar recordings" out for our team  @daibhin @pauldambra 
   - start generating clusters from recording embeddings for our team
   - figure out the UI for how to surface this to folk
   - measure performance needs of crunching all of this data 

### side quests

- contribute the network capture back to rrweb - so they don't make one that doesn't work for us - @pauldambra
- network payload capture on onboarding? @pauldambra 
- ✅ adding billing limits to unsubscribe? @pauldambra 
    - already present
    - once controls are generally released we can prompt people the first month they start to pay maybe
- how much embedding/clustering can we do just with session recording metadata (when can we avoid Open AI) @daibhin 


## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red

- 🟡📱 happy android replay users
   - had feedback from 3 beta testers
   - 1 elected to keep it switched on
   - still looking for sweet spot between capturing right info and setting expectations
- 🟡 people love replay even more
   - still more bug reports than we'd like
   - canvas replay is out
   - some rrweb bugs patched directly 
- 🟡 first class mobile support
   - not focussed here, focus is on replay
   - but making small pieces of progress
   - continuing to crank out improvements & features to mobile SDKs

### High priority


### Low priority / side quests

**Comment by robbie-c** [comment]
# Team Web Analytics

## Last sprint 
* Spent a lot of time on discussions and RFCs
* Most of the way to having a sessions table design and queries (see https://github.com/PostHog/posthog/pull/20462 and https://posthog.slack.com/archives/C05LJK1N3CP/p1708536198957149 )

## Hang over items from previous sprint
* Sessions table not shipped yet

## OKR
1. Finish the [beta to-do list](https://github.com/PostHog/posthog/issues/18547)
2. Find 5 high-ICP reference customers

### High priority
* Performance and architecture

### Low priority
* polish and be responsive to support
* Add ability to separate sites (e.g. Sites scene or saved filters)

**Comment by tiina303** [comment]
# Team Pipeline

Off: Tiina (2 weeks), Brett (1 week), Ted (3 days)
Support: Ted & Xavier

## Retro

### High priority

- [x] PoE: Write path for the new distinctId->PersonId overrides - Ted
- [x] Backfill implemented and verified for teams with large number of distinct IDs - Ted
- [ ] Ran squash for a single team - Tomas
     - tried, failed in the last step, likely will be done by end of week
- [x] Batch Export for cloud to cloud migrations - Brett
- [ ] pipeline UI available for beta testers - Tiina
     - made some progress, but got sidetracked with incidents and other requests

### Low priority / side quests

- [ ] Person property updates don't slow down ingestion - Brett
- [ ] backfill topic incident follow-ups - Tiina

## OKR
✅=finished 🟢=on track to finish this quarter 🟡=might not finish 🔴=won't finish

- ✅ Wrapped up Webhooks Delivery system - Brett
- ✅ Wrapped up capture-rs posthog-js rollout - Xavier
- 🟢 Wrapped up New Pipeline UI - Tiina
- 🟢 Wrapped up PoE (with backfills and squashes) - Ted
- 🟢 Volume spikes on session reply do not hurt other users - Xavier
- 🟢 Person property updates don't slow down ingestion - Brett
- 🟢 Requested data deletion is complete and reliably executed going forward - Tiina

### High priority

- [ ] Validated PoE performance & potential optimizations - Ted
- [ ] Replay overflow implemented (blobby detects, updates redis zset, capture reroutes) - Xavier
- [ ] Improve visibility into pipeline - Brett

### Low priority / side quests

- [ ] Distinct ID tombstoning in PG to fix person reuse - Ted
- [ ] Incident follow-ups

**Comment by raquelmsmith** [comment]
# Team Growthi

## Retro
<details>
<summary>Retro items</summary>

Lots of things in progress / left over. Quota limiting, Teams plan.

</details>


## OKR
✅=finished 🟢=on track to finish this quarter 🟡=might not finish 🔴=won't finish
- 🟡 Improve activation rates for product analytics and session replay
  - ✅   Polish onboarding flow with Cory
  - 🟢  Run 1 experiment per week
  - 🟡 Do an average of 2 customer interviews per week
  - 🟡 Redefine and improve activation rates for PA & SR
- ⚪ Launch pricing for:
  - 🟢 Teams
  - 🟡 Product analytics 21M free

## This sprint
### High priority
- @raquelmsmith 
    - [ ] Teams plan - in progress
    - [ ] 21M events free plan - in progress - need to make RFC for billing, but can start work
        - unless we want to wait until non-identified events are done, but I don't think we need to
    - [ ] Rethink some activation metrics (remove team member invited, double check rates with diff activities) - in progress
    - [ ] Prep for Zach to start 🙌 
- @xrdt 
    - [ ] Don't drop data for 7 (or maybe 3) days until after it's done - in progress, almost done 
    - [x] Experiment - put people directly on insight creation page 
    - [ ] Experiment on wording for "upgrade" / "subscribe" / "add a credit card" on billing page 
    - [ ] Let people self-serve to new plans, like the new replay plan so they can get access to the controls - start


### Low priority / side quests

- [ ] Write two articles for Andy
- [ ] Expired sessions should redirect users to the login page rather than popping lots of error toasts.
- [ ] Overlap in billing gauge projected / current / billing limit lines. https://posthoghelp.zendesk.com/agent/tickets/10070 (moved to PostHog: https://us.posthog.com/project/2/support/tickets/12692)

**Comment by neilkakkar** [comment]
# Team Feature Success

**Support hero:** @jurajmajerik
Days off:
Juraj: 3 days
Neil: 1 days

## Retro

- Experiments UI create & view overhaul - @jurajmajerik -> Mostly roll over, working on create UI currently. Paywall on experiments took a while to align on + finally remove.
- Explore experiment metrics easily in insights - @jurajmajerik -> Will roll over
- ~~Event based triggers for surveys (RFC here: https://github.com/PostHog/product-internal/pull/556/ )- @liyiy~~
- Find a good flow for restricting users to exposed users in secondary metrics for experiments - @neilkakkar -> Will roll over. Was unexpectedly sick for 4 days, release condition refactor took much longer, additional survey support, and incident happened.
- Add property filters to event matching in cohorts - @neilkakkar -> will roll
- Filter out internal users bug in experiments - @neilkakkar -> will roll


## Hang over items from previous sprint

---

## OKRs

1. 🔴 Doubling down on surveys
2. 🟡 Doubling down on experiments

### High priority

- Experiments UI create & view overhaul - @jurajmajerik
- Fix common support issues with surveys, so support is minimal here - @jurajmajerik 
- Address any incident remediation issues - @neilkakkar 
- Filter out internal users bug in experiments - @neilkakkar
- Local evaluation of cohorts in PHP SDK - @neilkakkar 


### Low priority / side quests

- Explore experiment metrics easily in insights - @jurajmajerik 
- Find a good flow for restricting users to exposed users in secondary metrics for experiments - @neilkakkar
- Add property filters to event matching in cohorts - @neilkakkar
- Temporal queues for feature success - @neilkakkar

**Comment by benjackwhite** [comment]
# Team Infra

**Support hero:** @daniel

## Hang over items from previous sprint

1. 🟢 Finalise `charts` movement and get Argo CD canary deploys working @danielxnj 
  - Charts are fully in `charts` - no more `charts-clickhouse` 
2. 🟡 Rollout all clients to new capture endpoints
  - JS is rolled out (us.i.posthog.com)
  - Remaining issue was `array.js` which we will solve with a new [redirect approach](https://github.com/PostHog/product-internal/issues/555)
  - Encountered more issues with IP resolution
3. 🟢 Mr Blobby -> EFS
  - Running in US and EU
  - Virtually no rebalancing costs :shocked_pikachu:
  - Just need to plan the switchover date and ensure autoscaling is in place


## OKR

1. 💪 Deploy with confidence 🟡
1. 💰 Keep costs under control ✅
1. 👀 More Observability 🔴
1. 🎯 Soc2 ✅

### High priority

1. Rollout redirect approach for static assets @danielxnj
2. Get everything onto contour ingress so we can offer Pipeline a way of gradually rolling out capture
3. Job spec out for SRE hire @benjackwhite 
4. Rollout Mr Blobby @benjackwhite 


### Low priority / side quests

**Comment by EDsCODE** [comment]
# Team Data Warehouse

### OKR Q1 2024

### Objective

**Get EU moved over to new cluster**

* **Key Results:**
    * Terraform, ansible, data migrated over, backups, housewatch
    * Deprecate old cluster

**ByConity evaluation**

* **Key Results:**
    * Load up with all clickhouse data and benchmark

**5 reference customers with qualities:**

* **Key Results:**
    * Paying a reasonable amount
    * Tell us they’re delighted
    * Not a big edge case
    * Using our in house data import pipeline
    * Weekly queries on a data warehouse connected table
    * Saving queried tables to insights?

**Data Warehouse queries are compatible with insights hogQL**

 

### Retro

**Product**
- [ ] @EDsCODE data warehouse data in trends
- [ ] @Gilbert09 View/table linking precursor to linking data warehouse data to person models

**Infra**
@fuziontech is out for 1/2 of next sprint
- [ ] Adapt current Ansible playbooks to new cluster configs starting with Coordinator
- [ ] TF Coordinator and have at least 2 running behind a load balancer

### High priority

**Product**
- [ ] @EDsCODE data warehouse data in trends breakdown frontend
- [ ] @EDsCODE Restore row limit tracking for all integrations
- [ ] @EDsCODE Reach out and do user research on current usage (the goal is to figure out what's needed to increase usage)
- [ ] @Gilbert09 Ship MVP: View/table linking precursor to linking data warehouse data to person models. 

**Infra**
